### PR TITLE
Decompile dScene_c

### PR DESCRIPTION
**Pull request by RootCubed**
Also decompile sPhase_c and update the docs for enum naming conventions

### CODE REVIEW

File: include/dol/bases/d_scene.hpp [hunk 1]
```diff
@@ -1,7 +1,62 @@
...
+    /// @brief Code to be executed after ::create.
+    /// @details Appends a LASTACTOR to the end of this scene's actors.
+    virtual void postCreate(fBase_c::MAIN_STATE_e status);
+
+    virtual int preDelete(); ///< See dBase_c::preDelete.
+    virtual void postDelete(fBase_c::MAIN_STATE_e status); ///< See dBase_c::postDelete.
+
+    /// @brief Code to be executed before ::execute.
+    /// @details If ::m_nextScene is set, prepares to transition to the next state.
+    /// Once all initial child bases have been added, clear the CONNECT and EXECUTE flag.
```
Comment (RootCubed) [inline include/dol/bases/d_scene.hpp]:
```suggestion
    /// Once all initial child bases have been added, enable the execute and draw step on this base and its children.
```

File: include/dol/bases/d_s_restart_crsin.hpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#pragma once
+#include <dol/bases/d_scene.hpp>
+
+class dScRestartCrsin_c : public dScene_c {
+public:
+    static void startTitle(unsigned char, bool);
+    static void reStartPeachCastle();
+};
```
Comment (RootCubed) [inline include/dol/bases/d_s_restart_crsin.hpp]: Missing terminating newline.

File: include/dol/bases/d_scene.hpp [hunk 3]
```diff
@@ -1,7 +1,62 @@
...
+class dScene_c : public dBase_c {
 public:
-    static ProfileName m_nowScene; ///< The profile name of the current scene.
+    dScene_c();
+    ~dScene_c();
+
+    virtual int preCreate(); ///< See dBase_c::preCreate.
+
+    /// @brief Code to be executed after ::create.
+    /// @details Appends a LASTACTOR to the end of this scene's actors.
```
Comment (RootCubed) [inline include/dol/bases/d_scene.hpp]:
```suggestion
    /// @details Appends a LASTACTOR child base to this scene.
```

File: include/dol/sLib/s_Phase.hpp [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+/// @brief A phase is a list of methods to be called in order.
+class sPhase_c {
+public:
+    /// @brief Return value of a phase method and ::callMethod.
+    /// @note Unofficial name.
+    enum CHAIN_RESULT_e {
```
Comment (RootCubed) [inline include/dol/sLib/s_Phase.hpp]: Rename this to `METHOD_RESULT_e`

File: source/dol/framework/f_base.cpp [hunk 5]
```diff
@@ -170,8 +170,8 @@ int fBase_c::execute() {
 }
 
 int fBase_c::preExecute() {
-    // Can only execute if not creating or deleting
-    if (mDeleteRequested || isProcessFlag(fManager_c::PROC_FLAG_CREATE)) {
+    // Can only execute if not creating or execution is disabled
```
Comment (RootCubed) [inline source/dol/framework/f_base.cpp]:
```suggestion
    // Can only execute if not deleting or execution is disabled
```

File: source/dol/framework/f_base.cpp [hunk 6]
```diff
@@ -230,19 +230,19 @@ int fBase_c::connectProc() {
         }
 
     } else {
-        // Copy over the process flags from the parent, if any
+        // Disable execute and draw step if the parent does so too
```
Comment (RootCubed) [inline source/dol/framework/f_base.cpp]:
```suggestion
        // Disable execute and/or draw step if the parent does so too
```

File: source/dol/framework/f_base.cpp [hunk 7]
```diff
@@ -42,14 +42,14 @@ fBase_c::fBase_c() :
         mMng.mDrawNode.mNewOrder = drawOrder;
     }
 
-    // Use the process flags from the parent, if set
+    // Disable execute and draw step if the parent does so too
```
Comment (RootCubed) [inline source/dol/framework/f_base.cpp]:
```suggestion
    // Disable execute and/or draw step if the parent does so too
```

File: include/dol/bases/d_scene.hpp [hunk 8]
```diff
@@ -1,7 +1,62 @@
...
+    /// @brief Sets the duration of the next fade-in transition to @p length.
+    static void setFadeInFrame(unsigned short length);
+    /// @brief Sets the duration of the next fade-out transition to @p length.
+    static void setFadeOutFrame(unsigned short length);
+    /// @brief Sets the duration of the next fade-in and fade-out transitions to @p length.
+    static void setFadeInOutFrame(unsigned short length);
+
+    sPhase_c *mpPhase; ///< The phase to be completed before ::preExecute succeeds.
+
+    static u32 mPara; ///< The parameter value for the next scene.
```
Comment (CLF78) [inline include/dol/bases/d_scene.hpp]:
```suggestion
    static u32 mPara; ///< The parameters for the next scene.
```

**Review by CLF78 — CHANGES_REQUESTED** [review]

**Review by CLF78 — APPROVED** [review]

**Pull request merged**